File: PYTHON/EJERCICIOS/BUCLES/Ejercicio22-ver-orden-lista.py
""""
EJERCICIO DE LOGICA DE PROGRAMACION
Hacer un programa que verifique en estado
(Descendente,Ascendente,Desordenada) 
esta un lista de numeros enteros.
"""
# Creamos una lista 
lista_numeros = [1,4,2,7,4,2]
# variables bandera
ascendente = False
descendente = False
# recoremos la lista 
for i in range(len(lista_numeros)-1):
    # si el numero en la posion i es mayor a la posion siguiente
    # cambia el valor de la variable
    if lista_numeros[i] > lista_numeros[i+1]:
        descendente = True
    else:
        ascendente = True
# mostramos el estado
if ascendente and descendente:
     print("La lista esta desordenada")
elif ascendente:
     print("La lista esta en forma ascendente")
else:
     print("La lista esta en forma descendente")

File: JAVA/EJERCICIOS/BUCLES/Ejercicio22VerOrdenLista.java
package JAVA.EJERCICIOS.BUCLES;
/*
EJERCICIO DE LOGICA DE PROGRAMACION
Hacer un programa que verifique en estado(Descendente,Ascentdente
,Desordenada) esta un lista de numeros enteros.
 */
import java.util.Arrays;
import java.util.List;
public class Ejercicio22VerOrdenLista {
    public static void main(String[] args) {
        // cremos la lista 
        List<Integer> listaNumeros = Arrays.asList(1,5,4,2,7);
        // variables banderas
        boolean descendente = false;
        boolean ascendente = false;
        // recorremos la lista
        for(int i = 0; i < listaNumeros.size() - 1; i++){
            if (listaNumeros.get(i) > listaNumeros.get(i+1)) {
                descendente = true;
            } else {
                ascendente = true;
            }
        }
        // mostramos el estado 
        if (ascendente && descendente) {
            System.out.println("La lista esta desordenada");
        } else if (ascendente) {
            System.out.println("La lista esta en forma ascendente");
        } else {
            System.out.println("La lista esta en forma descendente");
        }
    }
}
